source/ast.rs

```diff
@@ -71,6 +71,117 @@ pub enum Literal {
     String(Vec<u8>)
 }
 
-/// An identifier represents any “entity name”.
+/// A name represents either a variable or an entity name.
 #[derive(Debug, PartialEq)]
-pub struct Identifier<'a>(pub &'a [u8]);
+pub enum Name<'a> {
+    /// A variable.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[macro_use]
+    /// # extern crate nom;
+    /// use nom::IResult::Done;
+    /// # extern crate taguavm_parser;
+    /// use taguavm_parser::rules::tokens::variable;
+    /// use taguavm_parser::ast::Name;
+    ///
+    /// # fn main () {
+    /// assert_eq!(
+    ///     variable(b"$foo"),
+    ///     Done(&b""[..], Name::Variable(&b"foo"[..]))
+    /// );
+    /// # }
+    /// ```
+    /// Note that the `$` is not present.
+    Variable(&'a [u8]),
```

Comment:
I feel like this doesn't really belong in here. Variables also have a name, but it's pretty different from the namespaced names the other parts are about...